### PR TITLE
Fix converging in modern chef 11 and chef 12

### PR DESCRIPTION
**Pull request by tbunnyman**
Also update the test suite

### CODE REVIEW

File: metadata.rb [hunk 1]
```diff
@@ -19,3 +19,4 @@
 supports "ubuntu"
 
 depends "build-essential", ">= 2.0.0"
```
Comment (martinisoft) [inline metadata.rb]: We can hard lock this to 4.0.0 to maintain Chef 11 and 12 compatibility. 5.0 breaks compatibility with Chef 11 so we can cross that bridge when we come to it.

File: metadata.rb [hunk 2]
```diff
@@ -19,3 +19,4 @@
...
+depends "apt"
```
Comment (martinisoft) [inline metadata.rb]: Would also suggest hard locking this to 3.0.0 for a similar reason of compatibility with Chef 11 and 12. We can look at a breaking change version which will alter this later.

File: .kitchen.yml [hunk 3]
```diff
@@ -1,23 +1,31 @@
 ---
 driver:
   name: vagrant
-  vagrantfile_erb: Vagrantfile.erb
```
Comment (martinisoft) [inline .kitchen.yml]: Why drop the support for vagrant-cachier? 😢

Comment (tbunnyman) [inline .kitchen.yml]: You should add vagrant-cachier in your global config, there is no good reason to set it up in the cookbook itself

File: metadata.rb (hunk 2 above)
Comment (tbunnyman) [inline metadata.rb]: I did this in berkshelf since this isn't actually a requirement of the cookbook but chef 11 itself

File: metadata.rb (hunk 1 above)
Comment (tbunnyman) [inline metadata.rb]: I did this in berkshelf since this isn't actually a requirement of the cookbook but chef 11 itself

File: .kitchen.yml (hunk 3 above)
Comment (martinisoft) [inline .kitchen.yml]: Can you update the README for that under Testing by removing it or suggesting how it can be done?

**Comment by martinisoft** [comment]
I am 👍  on this.

**Pull request merged**